### PR TITLE
Add Dockerfile, nginx config, and Makefile

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Add Dockerfile, nginx config, and Makefile for an example of how to deploy.

Assumes config are in an `env-config.js` file.

### Why
For deployment.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - && \
+    echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list && \
+    apt-get update && apt-get install -y nodejs yarn
+
+ADD . /app/src
+
+WORKDIR /app/src
+
+RUN yarn install
+RUN yarn build
```
Comment (fnando) [inline docker/Dockerfile:15]: Can we remove sourcemaps after building the app?

```
RUN rm build/static/{css,js}/*.map
```

Comment (leighmcculloch) [reply docker/Dockerfile:15]: Is it possible to make the yarn build process not generate them? The .js files reference the source maps, so ideally we just wouldn't generate them and they wouldn't be referenced with comments in the minified js files either.

Comment (fnando) [reply docker/Dockerfile:15]: Not with create-react-app. It's not a problem files referencing sourcemap files, as those are only loaded when someone opens the web dev tools. Then, those calls would return 404s, which is no big deal.

Comment (leighmcculloch) [reply docker/Dockerfile:15]: Gotcha. I'll add a rm of the maps then 👍.

Comment (leighmcculloch) [reply docker/Dockerfile:15]: We chatted offline and decided this doesn't really matter because this is open source.

**Pull request merged**